### ISSUE TITLE
Unable to add member to gitlab project based on `username` field

### ISSUE THREAD

**Issue by KyzEver** [python-gitlab/python-gitlab#3050]
## Description of the problem, including code/CLI snippet
I am unable to add a member to a project based on `username`; it will only accept `user_id`, though the gitlab supports them mutually exclusively: https://docs.gitlab.com/ee/api/members.html#add-a-member-to-a-group-or-project

## Expected Behavior
I should be able to add a member based on `username`.

## Actual Behavior
When I try and add someone based on `username`, I receive a `AttributeError: Missing attributes: user_id` exception.

```
def create_repo(area):
    proj = gl.projects.create({'name': area, 'namespace_id': GROUP_ID})
    proj.members.create({'username': USERNAME, 'access_level': gitlab.const.AccessLevel.MAINTAINER})
    return proj
```
Curiously, it does recognize that it should be mutually exclusive, however...

```
def create_repo(area):
    proj = gl.projects.create({'name': area, 'namespace_id': GROUP_ID})
    proj.members.create({user_id: None, 'username': USERNAME, 'access_level': gitlab.const.AccessLevel.MAINTAINER})
    return proj
```
`gitlab.exceptions.GitlabCreateError: 400: user_id, username are mutually exclusive`

## Specifications

  - python-gitlab version:
  ```
>>> gitlab.__version__
'4.8.0'
```
  - Gitlab server version (or gitlab.com): v17.5.2-ee

**Comment by JohnVillalovos** [comment]
Any chance you could add the traceback text?

**Comment by JohnVillalovos** [comment]
Oh. I think this might have been fixed in python-gitlab 4.11.0 https://github.com/python-gitlab/python-gitlab/releases/tag/v4.11.0

feat(api): add exclusive GET attrs for /projects/:id/members ([e637808](https://github.com/python-gitlab/python-gitlab/commit/e637808bcb74498438109d7ed352071ebaa192d5))

**Comment by nejch** [comment]
Yes, this was fixed in https://github.com/python-gitlab/python-gitlab/pull/2970. @KyzEver please upgrade and reopen if you still see any issues with the latest version.

**Comment by KyzEver** [comment]
Sorry I didn't follow-up with this -- yes I was mistakenly using an older version that didn't have this resolved :( apologies for the rookie mistake, I thought I was on the most recent version.
